internal/tools/cloudflare/dns.go

```diff
@@ -0,0 +1,164 @@
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	dnsRecords, err := c.cfClient.DNSRecords(ctx, zoneID, cf.DNSRecord{Name: customerDNSName})
+	if err != nil {
+		logger.WithError(err).Error("failed to get DNS Record ID from Cloudflare")
+		return "", err
+	}
+	if len(dnsRecords) == 0 {
+		logger.Info("Unable to find any DNS records in Cloudflare; skipping...")
+		return "", nil
+	}
+
+	return dnsRecords[0].ID, nil
+
+}
+
+// CreateDNSRecord creates a DNS record in the first given Cloudflare zone name of the list
+func (c *Client) CreateDNSRecord(customerDNSName string, zoneNameList []string, dnsEndpoints []string, logger logrus.FieldLogger) error {
+
+	if len(dnsEndpoints) == 0 {
+		return errors.New("no DNS endpoints provided for Cloudflare creation request")
+	}
+	dnsEndpoint := dnsEndpoints[0]
+	if dnsEndpoint == "" {
+		return errors.New("DNS endpoint was an empty string")
+	}
+
+	// Fetch the zone name for that customer DNS name
+	zoneName, found := c.getZoneName(zoneNameList, customerDNSName)
+	if !found {
+		return errors.Errorf("hosted zone for %q domain name not found", customerDNSName)
+	}
+
+	// Fetch the zone ID
+	zoneID, err := c.getZoneID(zoneName)
+	if err != nil {
+		logger.WithError(err).Error("failed to fetch Zone ID from Cloudflare")
+		return err
+	}
+
+	proxied := true
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	recordResp, err := c.cfClient.CreateDNSRecord(ctx, zoneID, cf.DNSRecord{
+		Name:    customerDNSName,
+		Type:    "CNAME",
+		Content: dnsEndpoint,
+		TTL:     1,
+		Proxied: &proxied,
+	})
+	if err != nil {
+		logger.WithError(err).Error("failed to create DNS Record at Cloudflare")
+		return err
+	}
+	fmt.Println(recordResp)
+
+	logger.WithFields(logrus.Fields{
+		"cloudflare-dns-value":    customerDNSName,
+		"cloudflare-dns-endpoint": dnsEndpoint,
+		"cloudflare-zone-id":      zoneID,
+	}).Debugf("Cloudflare create DNS record response: %v", recordResp)
+
+	return nil
+}
+
+// DeleteDNSRecord gets DNS name and zone name which uses to delete that DNS record from Cloudflare
+func (c *Client) DeleteDNSRecord(customerDNSName string, zoneNameList []string, logger logrus.FieldLogger) error {
+
+	// Fetch the zone name for that customer DNS name
+	zoneName, found := c.getZoneName(zoneNameList, customerDNSName)
+	if !found {
+		return errors.Errorf("hosted zone for %q domain name not found", customerDNSName)
+	}
+
+	// Fetch the zone ID
+	zoneID, err := c.getZoneID(zoneName)
+	if err != nil {
+		logger.WithError(err).Error("failed to fetch Zone ID from Cloudflare")
+		return err
+	}
+
+	recordID, err := c.getRecordID(zoneID, customerDNSName, logger)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get record ID from Cloudflare for DNS: %s", customerDNSName)
+		return err
+	}
+
+	// Unable to find any record, skipping deletion
+	if err == nil && recordID == "" {
```

Comment:
We return earlier when `err` is not nil so I think we can drop this check.
```suggestion
	if recordID == "" {
```